### PR TITLE
samples: mesh: build only on BLE platforms

### PR DESCRIPTION
**Pull request by nashif**

### CODE REVIEW

**Review by galak — COMMENTED** [review]
I was thinking of something similar.  Wondering what or how we describe platforms/boards that support BLE over UART.  Do we introduce a 'ble-uart' device type?

**Comment by nashif** [comment]
> I was thinking of something similar. Wondering what or how we describe platforms/boards that support BLE over UART. Do we introduce a 'ble-uart' device type?

I would not got into this level of details. Whether a board support ble over UART or SPI should be done in the board configuration. When building a BLE application this will be handled automatically. If a board such as the carbon is still not ready to use BLE in Zephyr (over SPI), we would just drop this as a supported feature.

File: boards/x86/arduino_101/arduino_101.yaml [hunk 1]
```diff
@@ -7,3 +7,4 @@ toolchain:
   - issm
 supported:
   - usb_device
+  - ble
```
Comment (Vudentz) [inline boards/x86/arduino_101/arduino_101.yaml]: I think bluetooth is preferable over ble, we don't really refer to ble in any of our APIs. BLE would suggest this works over single mode only, but mesh defines transports such as GATT Proxy which in theory works over BR/EDR aka. classic or dual-mode as well.

File: samples/bluetooth/mesh/sample.yaml [hunk 2]
```diff
@@ -1,14 +1,8 @@
 sample:
     description: TBD
-    name: TBD
+    name: BLE Mesh
```
Comment (Vudentz) [inline samples/bluetooth/mesh/sample.yaml]: Bluetooth mesh

File: samples/bluetooth/mesh_demo/sample.yaml [hunk 3]
```diff
@@ -1,14 +1,8 @@
 sample:
     description: TBD
-    name: TBD
+    name: BLE Mesh Demo
```
Comment (Vudentz) [inline samples/bluetooth/mesh_demo/sample.yaml]: Ditto.

**Comment by galak** [comment]
That all makes sense, how do we handle boards that allow plugging in bluetooth module over some pin header (like FRDM-K64F)?

File: samples/bluetooth/mesh_demo/sample.yaml [hunk 4]
```diff
@@ -1,14 +1,8 @@
...
+-   test:
         build_only: true
-        filter: CONFIG_SOC == "ia32"
-        tags: bluetooth
--   test_arm:
-        arch_whitelist: arm
-        build_only: true
-        platform_exclude: arduino_due
         tags: bluetooth
+        depends_on: ble
```
Comment (Vudentz) [inline samples/bluetooth/mesh_demo/sample.yaml]: bluetooth

**Review by galak — DISMISSED** [review]

**Review by Vudentz — CHANGES_REQUESTED** [review]

**Comment by nashif** [comment]
> That all makes sense, how do we handle boards that allow plugging in bluetooth module over some pin header (like FRDM-K64F)?

every board can add bluetooth shields or any shield that adds more features. There was a proposal to add such a layer on top of boards, that is where something like that would be added, i.e. Shield XYZ supporting bT would have this dependency, not the board.

**Review by galak — CHANGES_REQUESTED** [review]
It seems like we really should have a distinction between ble and bluetooth.  The test/sample might depend on either being supported, but I think its better tha the board yaml's stay more specific.

**Comment by jhedberg** [comment]
What's the difference between "ble" and "bluetooth"? Bluetooth Low Energy (BLE for short) is one type of Bluetooth. Bluetooth BR/EDR (aka Classic) is another type, however one which is not fully supported by Zephyr. So for most purposes, in a Zephyr context, BLE and Bluetooth are one and the same.

**Comment by galak** [comment]
what if we support bluetooth BR/EDR (aka Classic) in the future?

**Comment by nashif** [comment]
> what if we support bluetooth BR/EDR (aka Classic) in the future? Seems like saying ble and leaving things as Anas has them is correct if we don't support (Classic) right now on any platform.

we do actually have support to a certain level for classic in the tree, we just do not have a suitable platform that supports it, so ble vs br/edr makes more sense :)

**Review by Vudentz — DISMISSED** [review]
LGTM

**Comment by nashif** [comment]
going back to ble for the  case when we have hardware that support classic to be able to make a distinction.

**Pull request merged**